sphinx/ext/intersphinx.py

```diff
@@ -575,10 +575,12 @@ def get_inventory_and_name_suffix(self, name: str) -> tuple[str | None, str]:
 
     def get_role_name(self, name: str) -> tuple[str, str] | None:
         """Find (if any) the corresponding ``(domain, role name)`` for *name*.
-        
+
         The *name* can be either a role name (e.g., ``py:function`` or ``function``)
-        given as ``domain:role`` or ``role``, or its corresponding object name (in 
-        this case, ``py:func`` or ``func``) given as ``domain:objname`` or ``objname``.
+        given as ``domain:role`` or ``role``, or its corresponding object name
+        (in this case, ``py:func`` or ``func``) given as ``domain:objname`` or ``objname``.
+
+        If no domain is given, the 'std' domain is used.
```

Comment:
Technically, the 'std' domain is used if you also do not find a corresponding role for the domain (like, if you use ``py:ref``, the domain is 'py', but 'ref' does not exist, so it branches to the second `if`). Now, should `py:ref` resolve to `std:ref` then?

I'm not sure whether this was even intended in the first place, but maybe we should document this behaviour as well?

Reply:
yeh good point, I have a feeling this was not intended 🤨,
but ok, lets not back-break and just document

Reply:
done